core/src/main/java/com/graphhopper/reader/dem/HeightTile.java

```diff
@@ -77,6 +78,15 @@ void setHeights(DataAccess da) {
         this.heights = da;
     }
 
+    private short getHeightSample(int x, int y) {
+        // always keep in mind factor 2 because of short value
+        return heights.getShort(2 * (y * width + x));
+    }
+
+    private double linearInterpolate(double a, double b, double f) {
+        return a < MIN_ELEVATION ? b : b < MIN_ELEVATION ? a : (a + (b - a) * f);
```

Comment:
Why is this handling with MIN_ELEVATION necessary? What does this mean and what is the unit?

And IMO the method is not guarded if a and b are smaller than MIN_ELEVATION.

Reply:
My understanding is that voids in elevation data (in SRTM at least) show up as Short.MIN_VALUE, so the MIN_ELEVATION check is meant to catch that since otherwise we'd be comparing a double to a short.

My thought here was that we make ourselves a bit more robust to voids if when any of the 4 surrounding points that we are averaging are missing, we average the other points.

Reply:
Oh, indeed. You are right. This is quite surprising as we have no proper NaN handling downstream for these values. Probably we never stumbled over this as we use CGIAR that has filled the void. So yes, this is probably for another issue and good idea about the surrounding points.